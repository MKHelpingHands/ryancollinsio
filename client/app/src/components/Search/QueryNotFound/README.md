## QueryNotFound Component
A component that shows a fun emoji when no items found for sitewide search.
