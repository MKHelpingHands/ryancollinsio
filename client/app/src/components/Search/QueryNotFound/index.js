import React, { PropTypes } from 'react';

import styles from './index.module.scss';
import cssModules from 'react-css-modules';

const QueryNotFound = ({ itemName }) => (
  <div className={styles.noneFound}>
    <img className={styles.noneFoundImage} src="https://github.com/RyanCCollins/cdn/blob/master/misc/thinking-face.png?raw=true" />
    <div className={styles.noneFoundText}>
      <p>
        Sorry, but we couldn't find any {itemName} that matched your query.
      </p>
    </div>
  </div>
);

QueryNotFound.propTypes = {
  itemName: PropTypes.string.isRequired,
};

export default cssModules(QueryNotFound, styles);
