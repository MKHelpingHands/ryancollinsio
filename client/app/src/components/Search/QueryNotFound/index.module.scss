.noneFound {
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  flex-direction: column;
  padding: 30px;
}

.noneFoundImage {
  height: 100px;
  width: 100px;
}

.noneFoundText {
  margin-top: 40px;
  color: #455A64 !important;
  width: 400px;
}

.thinkingFace {
  font-size: 6rem;
  padding: 10px;
}
