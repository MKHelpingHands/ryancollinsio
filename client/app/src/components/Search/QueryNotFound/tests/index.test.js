import QueryNotFound from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<QueryNotFound />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <QueryNotFound />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
