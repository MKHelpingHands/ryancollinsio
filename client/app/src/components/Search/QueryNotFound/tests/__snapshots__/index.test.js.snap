exports[`<QueryNotFound /> should render with default props 1`] = `
<div>
  <img
    src="https://github.com/RyanCCollins/cdn/blob/master/misc/thinking-face.png?raw=true" />
  <div>
    <p>
      Sorry, but we couldn\'t find any 
       that matched your query.
    </p>
  </div>
</div>
`;
